ire/src/main/scala/hu/bme/mit/ire/Transactions.scala

```diff
@@ -0,0 +1,91 @@
+package hu.bme.mit.ire
+
+import hu.bme.mit.ire.datatypes.Tuple
+import hu.bme.mit.ire.messages.ChangeSet
+
+import scala.collection.mutable
+
+trait Transaction extends AutoCloseable {
+  override def close(): Unit
+
+  def add(pred: String, node: Tuple)
+
+  def remove(pred: String, node: Tuple)
+}
+
+class TransactionFactory(val messageSize: Int = 16) {
+  val subscribers = new mutable.HashMap[String, mutable.MutableList[(ChangeSet) => Unit]]
+  val usedIDs = new mutable.HashSet[Long]()
+  val idGenerator = new scala.util.Random
+
+  def subscribe(subscriber: Map[String, (ChangeSet) => Unit]) = {
+  for ((attribute, func) <- subscriber)
+    subscribers.getOrElseUpdate(attribute, mutable.MutableList()) += func
+  }
+
+  def newBatchTransaction(): BatchTransaction = {
+  new BatchTransaction()
+  }
+
+  def newContinousTransaction(): ContinuousTransaction = {
+  new ContinuousTransaction(messageSize)
```

Comment:
I guess the editorconfig support in Eclipse is broken for pretty much anything but Java. For example, it also messes up the Xtend files - see the indentation of https://github.com/FTSRG/ingraph/blob/master/ingraph-report/src/test/java/ingraph/report/tests/TckReportTest.xtend#L51